orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/serialize/DeserializeApplicationTask.groovy

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.serialize
+
+import com.netflix.spinnaker.orca.DefaultTaskResult
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.Task
+import com.netflix.spinnaker.orca.TaskResult
+import com.netflix.spinnaker.orca.clouddriver.KatoService
+import com.netflix.spinnaker.orca.clouddriver.tasks.AbstractCloudProviderAwareTask
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Component
+class DeserializeApplicationTask extends AbstractCloudProviderAwareTask implements Task {
+
+  @Autowired
+  KatoService kato
+
+  @Override
+  TaskResult execute(Stage stage) {
+    String cloudProvider = getCloudProvider(stage)
+    String account = getCredentials(stage)
+    def taskId = kato.requestOperations(cloudProvider, [[deserializeApplication: stage.context]])
+      .toBlocking()
+      .first()
+    Map outputs = [
+      "notification.type"       : "deserializeapplication",
+      "kato.last.task.id"       : taskId,
+      "deserialize.application" : stage.context.applicationName,
+      "deserialize.snapshot"    : stage.context.snapshotId,
```

Comment:
yeah, I'm not positive what it will look like now (might be a generation or a timestamp). I can change it to a more descriptive name when its fully implemented